### PR TITLE
perlPackages.TclpTk: init at 1.09

### PR DESCRIPTION
**Pull request by willcohen**
###### Description of changes

Adds `Tcl::pTk` perl package, with necessary dependencies included as well. Builds on darwin and linux, though GUI functionality of `Tcl::pTk` only tested on darwin.

Note to @agbrooks -- as you noted in #122472, my previous snippet for `perlPackages.Tcl` didn't work on linux. I had to update the `makeMakerFlags` to include tclsh and disable stubs. I still struggled with the tk issue noted there, and opted instead to add `Tcl::pTk` instead of `Tcl::Tk` for now.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by willcohen** [comment]
Result of `nixpkgs-review pr 175775` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>perl532Packages.Tcl</li>
    <li>perl532Packages.TclpTk</li>
    <li>perl532Packages.TextLevenshteinXS</li>
    <li>perl532Packages.TkToolBar</li>
    <li>perl532Packages.WebServiceValidatorHTMLW3C</li>
    <li>perl534Packages.Tcl</li>
    <li>perl534Packages.TclpTk</li>
    <li>perl534Packages.TextLevenshteinXS</li>
    <li>perl534Packages.TkToolBar</li>
    <li>perl534Packages.WebServiceValidatorHTMLW3C</li>
  </ul>
</details>

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 175775` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>bwidget</li>
    <li>perl532Packages.Tcl</li>
    <li>perl532Packages.TclpTk</li>
    <li>perl532Packages.TextLevenshteinXS</li>
    <li>perl532Packages.TkToolBar</li>
    <li>perl532Packages.WebServiceValidatorHTMLW3C</li>
    <li>perl534Packages.Tcl</li>
    <li>perl534Packages.TclpTk</li>
    <li>perl534Packages.TextLevenshteinXS</li>
    <li>perl534Packages.TkToolBar</li>
    <li>perl534Packages.WebServiceValidatorHTMLW3C</li>
  </ul>
</details>

**Review by zakame — APPROVED** [review]

**Review by stigtsp — COMMENTED** [review]
Hi there!

`TextLevenshteinXS` and `WebServiceValidatorHTMLW3C` do not seem to be dependencies of `TclpTk` afaik. Would it be better to set up separate PRs for those?

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.TclpTk perlPackages.TkToolbar perlPackages.Tcl bwidget

**Comment by willcohen** [comment]
@stigtsp the only consideration that's not fully resolved is the use of bwidget and tk in Tcl. Neither are actually hard dependencies of Tcl, but pTk projects may need both depending on the set of widgets they utilize (see #122472). Since this list could very well grow if/when more Tcl-dependent packages are added, an option might be to have just Tcl be vanilla with no additional tcl packages, and have something like TclFull be the growing list, since I'm not confident that these two additional packages are the only ones that might want to be added. 

The reason I didn't go down this route already is since no one has to date had a use for such a vanilla Tcl package for perl, so it seemed like it might be a premature optimization.

**Review by stigtsp — DISMISSED** [review]
LGTM, thanks!

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -21031,6 +21031,36 @@ let
...
+      sha256 = "sha256-LR+YBlKS9+W7mBBy9/EkAOjxGVVe4MC5zToPr/pXl24=";
+    };
+    propagatedBuildInputs = [
+      ClassISA
+      SubName
+      Tcl
+      TestDeep
+    ];
+    buildPhase = ''
+      echo $TCLLIBPATH
```
[the remark this answers is not in the document]
Comment (willcohen) [reply pkgs/top-level/perl-packages.nix]: Done with latest force push. Sorry for missing!

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.TclpTk perlPackages.TkToolbar perlPackages.Tcl

**Comment by willcohen** [comment]
Thank you very much!

**Pull request merged**